testutil/retry/retry.go

```diff
@@ -0,0 +1,92 @@
+// Package retry provides support for repeating operations in tests.
+//
+// A sample retry operation looks like this:
+//
+//   func TestX(t *testing.T) {
+//       for r := retry.OneSec(); r.Next(t.FailNow); {
```

Comment:
Hmm - it felt good to pass `t` in here, but this is pretty flexible and not much more complicated.